src/PowerShellEditorServices/Language/AstOperations.cs

```diff
@@ -55,88 +56,95 @@ internal static class AstOperations
             ILogger logger,
             CancellationToken cancellationToken)
         {
-            var type = scriptAst.Extent.StartScriptPosition.GetType();
-            var method =
+            if (!s_completionHandle.Wait(0))
+            {
+                return null;
+            }
+
+            try
+            {
+                var type = scriptAst.Extent.StartScriptPosition.GetType();
+                var method =
 #if CoreCLR
-                type.GetMethod(
-                    "CloneWithNewOffset",
-                    BindingFlags.Instance | BindingFlags.NonPublic);
+                    type.GetMethod(
+                        "CloneWithNewOffset",
+                        BindingFlags.Instance | BindingFlags.NonPublic);
 #else
-                type.GetMethod(
-                    "CloneWithNewOffset",
-                    BindingFlags.Instance | BindingFlags.NonPublic,
-                    null,
-                    new[] { typeof(int) }, null);
+                    type.GetMethod(
+                        "CloneWithNewOffset",
+                        BindingFlags.Instance | BindingFlags.NonPublic,
+                        null,
+                        new[] { typeof(int) }, null);
 #endif
 
-            IScriptPosition cursorPosition =
-                (IScriptPosition)method.Invoke(
-                    scriptAst.Extent.StartScriptPosition,
-                    new object[] { fileOffset });
-
-            logger.Write(
-                LogLevel.Verbose,
-                string.Format(
-                    "Getting completions at offset {0} (line: {1}, column: {2})",
-                    fileOffset,
-                    cursorPosition.LineNumber,
-                    cursorPosition.ColumnNumber));
-
-            CommandCompletion commandCompletion = null;
-            if (powerShellContext.IsDebuggerStopped)
-            {
-                PSCommand command = new PSCommand();
-                command.AddCommand("TabExpansion2");
-                command.AddParameter("Ast", scriptAst);
-                command.AddParameter("Tokens", currentTokens);
-                command.AddParameter("PositionOfCursor", cursorPosition);
-                command.AddParameter("Options", null);
-
-                PSObject outputObject =
-                    (await powerShellContext.ExecuteCommand<PSObject>(command, false, false))
-                        .FirstOrDefault();
-
-                if (outputObject != null)
+                IScriptPosition cursorPosition =
+                    (IScriptPosition)method.Invoke(
+                        scriptAst.Extent.StartScriptPosition,
+                        new object[] { fileOffset });
+
+                logger.Write(
+                    LogLevel.Verbose,
+                    string.Format(
```

Comment:
string interpolation `$"Getting completions at offset {fileOffset} (line: {cursorPosition.LineNumber}, column: {cursorPosition.ColumnNumber})"` would make it more readable

Reply:
It's hard to tell without hiding whitespace changes but most of this file is unaltered.  This is another place where I think it makes sense to wait until we can fix this correctly with Serilog.

Reply:
Theoretically, the Serilog change has already happened. I agree we should use Serilog's format string mechanism, but as is, we just use it internally and preserve the old interface.

Reply:
@rjmholt right, I had assumed we intended to write a custom `ITextFormatter` at some point to avoid building all those verbose strings if verbose logging wasn't enabled.  If that wasn't planned I'll see if I can whip something up after this PR merges.

Reply:
Was about to type out a long reply, but I'll open an issue to move the discussion there.